basetest.pm

```diff
@@ -17,6 +17,13 @@ use Mojo::File 'path';
 my $serial_file_pos = 0;
 my $autoinst_log_pos = 0;
 
+my $fail_severity = {
+    '' => 0,
+    'ok' => 100,
+    'softfail' => 200,
+    'fail' => 300
```

Comment:
The mapping needs a few more values, otherwise the PR looks good.

    select distinct(result) from job_modules;
       result   
    ------------
     none
     failed
     pass
     running
     passed
     skipped
     softfailed
     canceled

Reply:
I'm not sure if that result strings are used there, as basetest.pm will set `ok` while this here uses `passed`.
So there seems to be some kind of translation somewhere.

Reply:
here we go:
`lib/OpenQA/Schema/Result/JobModules.pm`:
```perl
sub update_result {
    my ($self, $r) = @_; 
    my $result = $r->{result} || 'none';
    $result =~ s,fail,failed,;
    $result =~ s,^na,none,;
    $result =~ s,^ok,passed,;
    $result =~ s,^unk,none,;
    $result =~ s,^skip,skipped,;
    $result = 'softfailed' if ($r->{dents} && $result eq 'passed');
    $self->update({result => $result});
    return $result;
}
```